packages/server/src/script/resendSignupEmails.ts

```diff
@@ -0,0 +1,136 @@
+        const numResends = info.resends ? info.resends.length : 0
+        const file = storageFileFromId(info.id.concat('_') + numResends)
+        await file.upload(pdfBuffer)
+
+        // Send faxes
+        let twilioResponses: TwilioResponse[] = []
+        if (method.faxes.length > 0) {
+            const uri = await file.getSignedUrl(24 * 60 * 60 * 1000)
+            const resposnes = await sendFaxes(uri, method.faxes)
+            twilioResponses = resposnes.map(({url, sid, status}) => ({url, sid, status}))
+
+            for (const twilioResponse of twilioResponses) {
+                if (twilioResponse.status != "queued") {
+                    console.log('Error sending faxes for ' + info.id + '. Exiting.')
+                    console.log(twilioResponse)
+                    return
+                }
+            }
+          }
+
+        // Update voter data in firestore.
+        const resend = { dateTime: new Date().getTime(), reason: resendReason }
+        await firestoreService
+            .db
+            .collection('StateInfo')
+            .doc(info.id)
+            .update({
+                resends: admin.firestore.FieldValue.arrayUnion(resend)
+            })
+    }
+}
+
+// Reads './data/conditions.json' and resends emails to all voter sign ups that satisfy conditions.
+const main = async () => {
+    const conditions = input['conditions']
+    // Ensure conditions in input JSON file has at least one entry.
+    if (Object.keys(conditions).length < 1) {
+        console.log('Must have at least one condition specified.')
+        return
+    }
+
+    // Fetch registrations.
+    let query = firestoreService.db.collection('StateInfo').where(conditions[0].field, conditions[0].op as WhereFilterOp, conditions[0].value)
+    for (const condition of conditions.slice(1, conditions.length)) {
+      query = query.where(condition.field, condition.op as WhereFilterOp, condition.value)
+    }
+    const registrations = await firestoreService.query<RichStateInfo>(query)
+    if (registrations == null || registrations.length == 0) {
+        console.log('No voters found for specified conditions.')
+        return
+    }
+
+    // Display registration previews.
+    const registrationsToDisplay = registrations.length > 10 ? registrations.slice(0, 10) : registrations
+    console.log('There are a total of %s entries for your query.  Here is a sample:', registrations.length)
+    for (const preview of registrationsToDisplay) {
+        const previewMap: Map<string, any> = new Map<string, any>()
+        previewMap.set('id', preview.id)
+        for (const condition of conditions) {
+            if (typeof condition.field !== 'string') {
+                throw new Error('Condition field is not a string.  Exiting.')
+            }
+            const stringCondition: string = condition.field
+            previewMap.set(condition.field, preview[stringCondition as keyof RichStateInfo])
```

Comment:
Hun?  What does this do?

Reply:
This shows the user a sample of 10 voters from their query in the form [id, field specified in condition1, field specified in condtion2, ...]